Spurious "nullary overrides nilary" warning involving Java

```java
interface A {
  int f();
}
```

```scala
trait B1 extends A { def f: Int }
trait C1 { def f = 2 }
class T1 extends B1 with C1

trait B2 extends A { def f: Int = 1}
trait C2 { self: B2 => override def f = 2 }
class T2 extends B2 with C2
```

There are spurious warnings at T1 and T2 with 2.13.12-bin-f7e30a6, the 2.13.12 RC:

```
➜ sandbox qsc A.java B.scala
B.scala:3: warning: method f in trait C1 defined without a parameter list overrides method f in trait B1 defined with a single empty parameter list [quickfixable]
class T1 extends B1 with C1
      ^
B.scala:7: warning: method f in trait C2 defined without a parameter list overrides method f in trait B2 defined with a single empty parameter list [quickfixable]
class T2 extends B2 with C2
      ^
2 warnings
```

I'll submit a PR.

--- a/src/compiler/scala/tools/nsc/typechecker/RefChecks.scala
+++ b/src/compiler/scala/tools/nsc/typechecker/RefChecks.scala
@@ -487,9 +487,9 @@ abstract class RefChecks extends Transform {
                   warnAdaptedNullaryOverride()
               }
               else if (member.paramLists.isEmpty) {
-                // NullaryOverrideAdapted is only added to symbols being compiled, so check for a mismatch
-                // if both symbols are mixed in from the classpath
-                if (!member.isStable && other.paramLists.nonEmpty && !exempted)
+                // Definitions that directly override get a parameter list and a `NullaryOverrideAdapted` attachment
+                // in Namers. Here we also warn when there's a mismatch between two mixed-in members.
+                if (!member.isStable && other.paramLists.nonEmpty && !exempted && !other.overrides.exists(javaDetermined))
                   warnAdaptedNullaryOverride()
               }
               else if (other.paramLists.isEmpty) {

--- a/test/files/pos/t12858/A.java
+++ b/test/files/pos/t12858/A.java
@@ -1,0 +1,3 @@
+interface A {
+  int f();
+}

--- a/test/files/pos/t12858/B.scala
+++ b/test/files/pos/t12858/B.scala
@@ -1,0 +1,9 @@
+// scalac: -Werror
+
+trait B1 extends A { def f: Int }
+trait C1 { def f = 2 }
+class T1 extends B1 with C1
+
+trait B2 extends A { def f: Int = 1}
+trait C2 { self: B2 => override def f = 2 }
+class T2 extends B2 with C2

